plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -99,8 +98,7 @@ func TestNewMetricsReaderInvalidAddress(t *testing.T) {
 	require.Error(t, err)
 	assert.Contains(t, err.Error(), "failed to initialize prometheus client")
 	assert.Nil(t, reader)
-	err = closer(context.Background())
-	require.NoError(t, err)
+	defer closer()
```

Comment:
reposting from https://github.com/jaegertracing/jaeger/pull/4595#pullrequestreview-1549000731

> closer should be invoked via defer **right after the object is created**. Otherwise `require` before it may panic and you would never reach this line.

Reply:
my bad. if panic, closer() won't be executed.